zerver/lib/cache.py

```diff
@@ -53,6 +55,22 @@ def remote_cache_stats_finish() -> None:
     remote_cache_total_time += time.time() - remote_cache_time_start
 
 
+def update_cached_cache_key_prefixes() -> list[str]:
+    # Clearing cache keys happens for all cache prefixes at once.
+    # Because the list of cache prefixes can only be derived from
+    # reading disk, we cache the list of cache prefixes, itself, in
+    # the cache.
+    found_prefixes: set[str] = set()
+    for deploy_dir in get_recent_deployments(None):
+        filename = os.path.join(deploy_dir, "var", "remote_cache_prefix")
+        if not os.path.exists(filename):
+            continue
+        with open(filename) as f:
+            found_prefixes.add(f.readline().removesuffix("\n"))
+    caches["default"].set("cache_key_prefixes", list(found_prefixes), timeout=60 * 60 * 24)  # 24h
+    return list(found_prefixes)
```

Comment:
A refinement we can probably do on top of this is having a post-deploy hook that checks using `lsof` or similar whether the old deployment directory is actually clear, and removes that cache key from here.

But we need to be really careful with doing that if we ever revert a deploy via a `/last/` symlink... we might just want a hard memcached flush when doing that, if we can find the right trigger for it? Not sure.

Can all be follow-ups if it's worth doing now.